src/openrc-shutdown/broadcast.c

```diff
@@ -179,24 +174,23 @@ void broadcast(char *text)
 		}
 
 		/*
-		 *	Open it non-delay
+		 * Open it non-delay
+		 * Set an alarm to generate EINTR in case we block.
 		 */
-		if (sigsetjmp(jbuf, 1) == 0) {
-			alarm(2);
-			flags = O_WRONLY|O_NDELAY|O_NOCTTY;
-			if (file_isatty(term) && (fd = open(term, flags)) >= 0) {
-				if (isatty(fd) && (tp = fdopen(fd, "w")) != NULL) {
-					fputs(line, tp);
-					fputs(text, tp);
-					fflush(tp);
-				}
+		alarm(2);
+		flags = O_WRONLY|O_NDELAY|O_NOCTTY;
+		if (file_isatty(term) && (fd = open(term, flags)) >= 0) {
+			if (isatty(fd) && (tp = fdopen(fd, "w")) != NULL) {
+				if ((fputs(line, tp) == EOF || fputs(text, tp) == EOF)
```

Comment:
There's a race here:

```c
fputs(..);
// alarm fires
fputs(..); // 2nd fputs, alarm is no longer present
```

But TBH I don't think I understand why it's blocking to begin with. Shouldn't O_NDELAY prevent that??